### ISSUE TITLE
Why there is no Batch Normalization or dropout layer in the model?

### ISSUE THREAD

**Issue by Pengchengu** [ellisdg/3DUnetCNN#39]
Hi @ellisdg ,
How did you solve the overfitting problem? Data augmentation may solve part of it, but from the network structure and my training results, there is still a severe overfitting problem.
P.S.  As the limitation of memory, i split volume into patches and then i have the room for increasing batch size. 
Cheers.

**Comment by ellisdg** [comment]
Hi @Pengchengu! Correct me if I am wrong, but my understanding was that batch normalization helps the network train faster, but doesn't necessarily prevent overfitting. Nevertheless, I have implemented batch normalization in [a separate branch](https://github.com/ellisdg/3DUnetCNN/tree/arbitrary_depth) that I am using to test new features.

I think one approach that could help overfitting is to make the data augmentation more robust. There should be [48 unique lossless 3D transformations](https://en.wikipedia.org/wiki/Octahedral_symmetry#The_isometries_of_the_cube) of a 3D cube of data. I haven't yet figured out how to write a function to perform these transformations. For further data augmentation, you could combine the lossless transformations with lossy transformations (such as zooming, shearing, etc.) to create a vast number of augmentation possibilities.

Are you using the BRATS data set? Also, do you have code for splitting the data up into patches? I have my own code written in the branch linked above, but I am running into issues during training.
